docs/calling-downstream-apis/custom-apis.md

```diff
@@ -0,0 +1,823 @@
+✅ Can't use HttpClientFactory  
+✅ Need to manually construct requests  
+
+**Example:**
+```csharp
+var authHeader = await _headerProvider.CreateAuthorizationHeaderForUserAsync(scopes);
+client.DefaultRequestHeaders.Add("Authorization", authHeader);
+```
+
+## Error Handling
+
+### IDownstreamApi Errors
+
+```csharp
+try
+{
+    var data = await _api.GetForUserAsync<MyData>("MyApi", "data");
+}
+catch (MicrosoftIdentityWebChallengeUserException ex)
+{
+    // User needs to consent
+    _logger.LogWarning(ex, "Consent required for scopes: {Scopes}", string.Join(", ", ex.Scopes));
+    throw; // Let ASP.NET Core handle consent flow
+}
+catch (HttpRequestException ex) when (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+{
+    return NotFound("Resource not found");
+}
+catch (HttpRequestException ex) when (ex.StatusCode == System.Net.HttpStatusCode.Unauthorized)
+{
+    return Unauthorized("API returned 401");
+}
+catch (Exception ex)
+{
+    _logger.LogError(ex, "API call failed");
+    return StatusCode(500, "An error occurred");
+}
+```
+
+### MicrosoftIdentityMessageHandler Errors
+
+```csharp
+try
+{
+    var response = await _httpClient.GetAsync("api/data");
+    
+    if (!response.IsSuccessStatusCode)
+    {
+        var error = await response.Content.ReadAsStringAsync();
+        _logger.LogError("API returned {StatusCode}: {Error}", response.StatusCode, error);
+        return StatusCode((int)response.StatusCode, error);
+    }
+    
+    var data = await response.Content.ReadFromJsonAsync<MyData>();
+    return Ok(data);
+}
+catch (HttpRequestException ex)
+{
+    _logger.LogError(ex, "HTTP request failed");
+    return StatusCode(500, "Failed to call API");
+}
+```
+
+## Best Practices
+
+### 1. Configure Timeout Values
+
+```csharp
+builder.Services.AddDownstreamApi("MyApi", options =>
+{
+    options.BaseUrl = "https://api.example.com";
+    options.HttpClientName = "MyApi";
+});
+
+builder.Services.AddHttpClient("MyApi", client =>
+{
+    client.Timeout = TimeSpan.FromSeconds(30);
+});
+```
+
+### 2. Use Typed Clients
+
+```csharp
+public interface IProductApiClient
+{
+    Task<List<Product>> GetProductsAsync();
+    Task<Product> GetProductAsync(int id);
+    Task<Product> CreateProductAsync(Product product);
+}
+
+public class ProductApiClient : IProductApiClient
+{
+    private readonly IDownstreamApi _api;
+    
+    public ProductApiClient(IDownstreamApi api)
+    {
+        _api = api;
+    }
+    
+    public Task<List<Product>> GetProductsAsync() =>
+        _api.GetForUserAsync<List<Product>>("MyApi", "products");
+    
+    public Task<Product> GetProductAsync(int id) =>
+        _api.GetForUserAsync<Product>("MyApi", $"products/{id}");
+    
+    public Task<Product> CreateProductAsync(Product product) =>
+        _api.PostForUserAsync<Product, Product>("MyApi", "products", product);
+}
+
+// Register
+builder.Services.AddScoped<IProductApiClient, ProductApiClient>();
+```
+
+### 3. Handle Rate Limiting
```

Comment:
I wouldn't recommend users do this manually.

https://learn.microsoft.com/en-us/dotnet/core/resilience/http-resilience?tabs=dotnet-cli

Most users should be able to use AddStandardResilienceHandler with options to fit their needs.